src/SIM/SIMbase.C

```diff
@@ -2563,9 +2544,11 @@ bool SIMbase::injectPatchSolution (Vector& sol, const Vector& vec,
       nndof != getNoFields(basis) && this->getNoFields(2) > 0) {
     int key = basis << 16 + nndof;
     if (addMADOFs.find(key) == addMADOFs.end())
-      addMADOFs[key] = MADof(myModel, this->getNoNodes(true), basis, nndof);
+      addMADOFs[key] = std::move(setupAdditionalMADOF(myModel,
```

Comment:
har ingenting å si for ytelse å gjøre det på gamlemåten. det er kun syntax - du kan returnere by value uten at det koster en kopi. bruk av return vs å sende en referanse er litt mer intuitivt til tider, that's all.